homeassistant/components/motioneye/__init__.py

```diff
@@ -164,13 +164,14 @@ async def async_update_data() -> dict[str, Any] | None:
         CONF_COORDINATOR: coordinator,
     }
 
-    current_cameras: set[tuple[str, str, int]] = set()
+    current_cameras: set[tuple[str, str]] = set()
     device_registry = await dr.async_get_registry(hass)
 
     @callback
     def _async_process_motioneye_cameras() -> None:
         """Process motionEye camera additions and removals."""
-        inbound_camera: set[tuple[str, str, int]] = set()
+        inbound_camera: set[tuple[str, str]] = set()
+        assert coordinator.data is not None
```

Comment:
Please rebase on top of https://github.com/home-assistant/core/pull/49827 where this assert can be removed.

Reply:
FYI: If this assert is removed, mypy produces:

```
homeassistant/components/motioneye/__init__.py:174: error: Unsupported right operand type for in ("Optional[Dict[str, Any]]")  [operator]
homeassistant/components/motioneye/__init__.py:177: error: Value of type "Optional[Dict[str, Any]]" is not indexable  [index]
```

Is this what you expected?

Reply:
Didn't notice that `async_update_data` returns optional. Then you need to properly handle case when it returns `None`.